lib/ansible/modules/packaging/os/snap.py

```diff
@@ -141,33 +142,42 @@ def remove_snap(module, snap_name):
     rc, out, err = module.run_command(cmd, check_rc=False)
 
     if rc == 0:
-        # snap has been removed
-        module.exit_json(msg="The snap '%s' has been removed" % str(snap_name), changed=True)
+        # Snap has been removed
+        return True
     else:
         # Something went wrong
         module.fail_json(msg=err)
 
 
 def main():
     module_args = dict(
-        name=dict(type='str', required=True),
+        name=dict(type='list', required=True),
         state=dict(type='str', required=False, default='present', choices=['absent', 'present']),
         classic=dict(type='bool', required=False, default=False)
     )
     module = AnsibleModule(argument_spec=module_args)
-    snap_name = module.params['name']
+    snap_names = module.params['name']
     state = module.params['state']
 
-    # Test if snap exists
+    # Check if snaps are valid
+    for snap_name in snap_names:
     rc, out, err = snap_exists(module, snap_name)
-    if rc:
+        if rc != 0:
         module.fail_json(msg="No snap matching '%s' available" % str(snap_name))
 
+    # Apply changes to the snaps
+    exit_changed = False
+    for snap_name in snap_names:
     if state == 'present':
-        install_snap(module, snap_name)
+            r_changed = install_snap(module, snap_name)
     elif state == 'absent':
-        remove_snap(module, snap_name)
+            r_changed = remove_snap(module, snap_name)
 
+        # If we modified one of the snaps, change the task `changed` return code to true
+        if r_changed == True and exit_changed == False:
```

Comment:
When you explicitly compare to `None`, `True` or `False` use `is` operator.
But in this case you don't need that.
you could do `if r_changed and not exit_changed:`

Reply:
Fixed in https://github.com/ansible/ansible/commit/273f5476d89538f8b814f1379eeef57aceb2b0b9

I hesitated to use `if r_changed and not exit_changed:` but yeah, I will modify that too